static/main.js

```diff
@@ -273,6 +274,26 @@ function initPolicies(options) {
     }
 }
 
+function filterComponentState(config, keysToRemove) {
+    function filterComponentStateImpl(component) {
+        if (component.content) {
+            for (var i = 0; i < component.content.length; i++) {
+                filterComponentStateImpl(component.content[i], keysToRemove);
+            }
+        }
+
+        if (component.componentState) {
+            Object.keys(component.componentState)
+                .filter(function (key) { return keysToRemove.includes(key); })
+                .forEach(function (key) { delete component.componentState[key]; });
+        }
+    }
+
+    config = cloneDeep(config);
```

Comment:
For objects such as our state, isn't it smaller to do `JSON.parse(JSON.stringify(config))`?

Reply:
I didn't do any of my own measurements, but the general consensus on some stack overflow posts was that json stringify/parse was slower than other options, and this code will execute quite frequently.

Reply:
Huh, I thought the opposite was true! Good to know then :)